components/org.wso2.carbon.identity.workflow.core/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/approval/v1/impl/MeApiServiceImpl.java

```diff
@@ -32,24 +33,39 @@
 public class MeApiServiceImpl extends MeApiService {
 
     @Autowired
-    UserApprovalService userApprovalService;
+    ApprovalEventService approvalEventService;
+
 
     @Override
     public Response getApprovalTaskInfo(String taskId) {
 
-        return Response.ok().entity(userApprovalService.getTaskData(taskId)).build();
+        return Response.ok().entity(approvalEventService.getTaskData(taskId)).build();
     }
 
     @Override
     public Response listApprovalTasksForLoggedInUser(Integer limit, Integer offset, List<String> status) {
 
-        return Response.ok().entity(userApprovalService.listTasks(limit, offset, status)).build();
+        return Response.ok().entity(approvalEventService.listTasks(limit, offset, status)).build();
     }
 
     @Override
     public Response updateStateOfTask(String taskId, StateDTO nextState) {
 
-        userApprovalService.updateStatus(taskId, nextState);
+        org.wso2.carbon.identity.workflow.engine.dto.StateDTO nextStateDTO = new org.wso2.carbon.identity.workflow.engine.dto.StateDTO();
+        if (nextState.getAction() == StateDTO.ActionEnum.APPROVE){
+            nextStateDTO.setAction(org.wso2.carbon.identity.workflow.engine.dto.StateDTO.ActionEnum.APPROVE);
+            approvalEventService.updateStatus(taskId, nextStateDTO);
+        } else if (nextState.getAction() == StateDTO.ActionEnum.REJECT){
+            nextStateDTO.setAction(org.wso2.carbon.identity.workflow.engine.dto.StateDTO.ActionEnum.REJECT);
+            approvalEventService.updateStatus(taskId, nextStateDTO);
+        } else if (nextState.getAction() == StateDTO.ActionEnum.RELEASE){
+            nextStateDTO.setAction(org.wso2.carbon.identity.workflow.engine.dto.StateDTO.ActionEnum.RELEASE);
+            approvalEventService.updateStatus(taskId, nextStateDTO);
+        } else if (nextState.getAction() == StateDTO.ActionEnum.CLAIM){
+            nextStateDTO.setAction(org.wso2.carbon.identity.workflow.engine.dto.StateDTO.ActionEnum.CLAIM);
+            approvalEventService.updateStatus(taskId, nextStateDTO);
```

Comment:
`approvalEventService.updateStatus(taskId, nextStateDTO);` This part is repetitive. Let's write a private method for the conversion logic and call the `approvalEventService.updateStatus(taskId, nextStateDTO);` at the end.